polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -302,6 +308,16 @@ impl Metrics {
 			metrics.unapproved_candidates_in_unfinalized_chain.set(count as u64);
 		}
 	}
+
+	pub fn observe_assignment_gathering_time(&self, stage: usize, elapsed_as_millis: usize) {
+		if let Some(metrics) = &self.0 {
+			let stage_string = stage.to_string();
+			metrics
+				.assignments_gathering_time_by_stage
+				.with_label_values(&[if stage < 10 { stage_string.as_str() } else { "inf" }])
```

Comment:
It's unclear for me from reading the code how many stages can there be. The comments describe only 2 stages: 0 and 1. Is there more and how are they calculated then?

Reply:
In practice this is bounded by nDelayTranches which on kusama is 89, so you could have up to 89 stages if you are on the path were at each tranche you have only one no-show.  

Now, because each new label value(stage) is  a new entry for the metric in prometheus having extra 89 entries might put unnecessary pressure on the metrics infrastracture, so I just bounded it to just 10 and everything else gets aggregated `inf`.

Bare in mind that getting to the new stage at least no-show(3 slots) period have to has passed so each new stage would add you 3 blocks to the finality lag, so in this case reaching to 10th stage we would already have a finality lag of 30 blocks(not the state we want to be :D).

Reply:
nit:
```
let stage_string = if stage < 10 { stage.to_string().as_str() } else { "inf" };
```

Could you leave a short comment with the explanation why here is 10?

Reply:
Sure.